src/atc/etl/loader.py

```diff
@@ -12,15 +14,29 @@ class Loader(EtlBase):
     and does not consume or change it.
     """
 
-    def __init__(self):
-        super().__init__()
```

Comment:
The changes to the loader does not change its core logic. The code added allows the loader to call either save or save_many dependent on the dataset_input_key(s) provided. If no keys are provided it will resort to do what it has always done - checking the length of the input dataset and use this to evaluate wether calling save or save_many.
The downside to building an extended loader is that it will act entirely as the current loader - bar the ability to handle keys. Further, it would require loaders (such as the delta loader) to have two classes "per type of loader" (i.e., it would require a delta loader for the normal loader and a delta loader for the extended loader) - and they would be almost identicial except for the handling of dataset keys.